clang/test/CIR/Lowering/brcond.c

```diff
@@ -0,0 +1,10 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -fclangir-mem2reg -emit-llvm %s -o %t.ll
```

Comment:
We should probably also validate the produced IR, instead of just checking for the absence of a crash.

Someone more experienced would know if it's better to have this as a C test or a CIR test.

Reply:
That makes sense. I have added a CIR test to validate the produced IRs.

Reply:
Thanks for adding the extra CIR test file, but for this specific PR it's fine if we only test things directly here, given the source is simple to write. See `clang/test/CIR/CodeGen/abstract-cond.c` for how to use filecheck with regexes for the SSA values.